addons/website/models/ir_qweb.py

```diff
@@ -111,6 +112,52 @@ def _post_processing_att(self, tagName, atts):
         if not website:
             return atts
 
+        # If the cookie banner is activated, 3rd-party embedded iframes should
+        # be controlled. 'domains' is a watchlist on the iframe's src itself,
+        # 'classes' is a watchlist on container elements in which the iframes
+        # are/could be built on the fly client-side for some reason.
+        cookiesWatchList = {
+            # Checking 2nd-level domains mainly for Google (numerous subdomains
+            # & top-level domains, it would be impossible to check them all).
+            'domains': set([
+                "youtu", "youtube", "youtube-nocookie",  # "youtu.be", "youtube.com", "youtube-nocookie.com",
+                "instagram", "instagr", "ig",  # "instagram.com", "instagr.am", "ig.me",
+                "vimeo",  # "player.vimeo.com", "vimeo.com",
+                "dailymotion", "dai",  # "dailymotion.com", "dai.ly",
+                "youku",  # "player.youku.com", "youku.com",
+                "tudou",  # "tudou.com",
+                "google",  # "google.com", "google.be", "maps.google.com", ...
+                "facebook", "fb",  # "facebook.com", "fb.com", "fb.me", "fb.watch",
+                "tiktok",  # "tiktok.com",
+                "x", "twitter", "t",  # "x.com", "twitter.com", "t.co",
+            ]),
+            'classes': set([
+                's_map',
+                's_instagram_page',
+                'o_facebook_page',
+                'o_background_video',
+            ]),
+        }
+        if (
+            website.cookies_bar
+            and not request.env['ir.http']._is_allowed_cookie('optional')
```

Comment:
There is an issue here actually.

You can try:
- Enable the cookies setting
- Drag & Drop a snippet and add a YT video
- Visit the page as public user, you will see the new "To see this interactive.." banner since you did not accept cookies yet
- Now accept cookies, the video will show, which is good
- Now reload the pages multiple times, you will see that it sometimes still shows the "To see this interactive.." banner. This is because each odoo worker (4 on runbot I guess?) has its own cache and some will still have the `data-need-cookies-approval` in the DOM I guess.

See https://app.screencastify.com/v3/watch/syyMLXtgwtfIN0LmNaIW

That's because the `_is_allowed_cookie` is user dependent (it's partly based on reading the request cookies to see what the user acceptes or not), while the post-processing and page rendering should not be.

Could you investigate if that's indeed that part which is problematic? I am not sure what would be the best solution, I can have a deeper look if you want, but some possibilities to maybe play with:
- `_get_template_cache_keys` add something inside it
- just check `website.cookies_bar` settings here instead of `_is_allowed_cookie` and manage in JS the rest